index.js

```diff
@@ -137,6 +141,33 @@ WPCOM.prototype.sendRequest = function (params, query, body, fn) {
   return sendRequest.call(this, params, query, body, fn)
 };
 
+WPCOM.prototype.wpPromise = function ( callback, params, timeout ) {
+  var timeout = timeout || DEFAULT_ASYNC_TIMEOUT,
+      timer;
+
+  var action = new Promise( function ( resolve, reject ) {
```

Comment:
I'm pretty sure the tests are failing because CirciCI is running an old version of something that doesn't support Promises. It should work if we update the appropriate entry in `package.json`.

The promise gets created when you call `WPCOM.wpPromise()` and it returns the new Promise. Since it's passing the tests on my machine and the PR isn't finished, I didn't think it was a big deal.